internal/users/tempentries/groups.go

```diff
@@ -0,0 +1,179 @@
+package tempentries
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ubuntu/authd/internal/log"
+	"github.com/ubuntu/authd/internal/users/localentries"
+	"github.com/ubuntu/authd/internal/users/types"
+)
+
+type groupRecord struct {
+	name   string
+	gid    uint32
+	passwd string
+}
+
+type temporaryGroupRecords struct {
+	idGenerator   IDGenerator
+	registerMutex sync.Mutex
+	rwMutex       sync.RWMutex
+	groups        map[uint32]groupRecord
+	gidByName     map[string]uint32
+}
+
+func newTemporaryGroupRecords(idGenerator IDGenerator) *temporaryGroupRecords {
+	return &temporaryGroupRecords{
+		idGenerator:   idGenerator,
+		registerMutex: sync.Mutex{},
+		rwMutex:       sync.RWMutex{},
```

Comment:
We normally call them just `Mu` in the codebase here, so can shorten the suffixes

Reply:
Yeah, I've seen that. I thought `rwMutex` would be a bit easier to understand than `rwMu`, but I don't care much, so I'll change it.